### PR TITLE
[SPARK-34558][SQL][FOLLOWUP] Do not fail Spark startup with a broken FileSystem

### PR DESCRIPTION
**Pull request by cloud-fan**
### What changes were proposed in this pull request?

This is a followup of https://github.com/apache/spark/pull/31671

https://github.com/apache/spark/pull/31671 qualifies the warehouse at the beginning, which may fail Spark startup if something goes wrong, like the underlying FileSystem can't be initialized.

This PR falls back to the old behavior and leave the warehouse path unqualified if qualifying fails.

### Why are the changes needed?

Fix a regression. It's important to be always able to start Spark app (e.g. spark-shell), so that we can debug.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

a new test case

### CODE REVIEW

**Comment by cloud-fan** [comment]
cc @yaooqinn

**Comment by yaooqinn** [comment]
For the thrift server, could this change lead an unhealthy server for end-users?

File: sql/core/src/test/scala/org/apache/spark/sql/SparkSessionBuilderSuite.scala [hunk 1]
```diff
@@ -472,4 +472,13 @@ class SparkSessionBuilderSuite extends SparkFunSuite with BeforeAndAfterEach wit
...
     }
   }
+
+  test("SPARK-33944: Create a working SparkSession with a broken FileSystem") {
+    val session = SparkSession.builder()
+      .master("local")
+      .config(WAREHOUSE_PATH.key, "my_dir")
+      .config("fs.file.impl", "non.existing.class")
+      .getOrCreate()
+    session.sql("SELECT 1").collect()
```
Comment (yaooqinn) [inline sql/core/src/test/scala/org/apache/spark/sql/SparkSessionBuilderSuite.scala:482]:
```suggestion
    val msg = "Cannot qualify the warehouse path, leaving it unqualified"
    val logAppender = new LogAppender(msg)
    withLogAppender(logAppender) {
      val session =
        SparkSession.builder()
          .master("local")
          .config(WAREHOUSE_PATH.key, "my_dir")
          .config("fs.file.impl", "non.existing.class")
          .getOrCreate()
      session.sql("SELECT 1").collect()
    }
    assert(logAppender.loggingEvents.exists(_.getRenderedMessage.contains(msg)))
```

**Comment by cloud-fan** [comment]
> For the thrift server, could this change lead to creating an unhealthy server for end-users?

At least it's the same as before, not worse.

**Review by yaooqinn — APPROVED** [review]

**Comment by yaooqinn** [comment]
LGTM

**Review by dongjoon-hyun — APPROVED** [review]

File: sql/core/src/test/scala/org/apache/spark/sql/SparkSessionBuilderSuite.scala [hunk 2]
```diff
@@ -472,4 +472,19 @@ class SparkSessionBuilderSuite extends SparkFunSuite with BeforeAndAfterEach wit
         expected)
     }
   }
+
+  test("SPARK-33944: Create a working SparkSession with a broken FileSystem") {
```
Comment (dongjoon-hyun) [inline sql/core/src/test/scala/org/apache/spark/sql/SparkSessionBuilderSuite.scala:476]: SPARK-34558?

Comment (dongjoon-hyun) [reply sql/core/src/test/scala/org/apache/spark/sql/SparkSessionBuilderSuite.scala:476]: Sorry for missing this. Let me make a quick follow-up.

**Pull request closed without merging**